edtLHD/src/main/java/fr/univtln/lhd/model/entities/dao/slots/SlotDAO.java

```diff
@@ -0,0 +1,130 @@
+package fr.univtln.lhd.model.entities.dao.slots;
+
+import fr.univtln.lhd.exceptions.IdException;
+import fr.univtln.lhd.model.entities.dao.DAO;
+import fr.univtln.lhd.model.entities.dao.Datasource;
+import fr.univtln.lhd.model.entities.slots.Classroom;
+import fr.univtln.lhd.model.entities.slots.Group;
+import fr.univtln.lhd.model.entities.slots.Slot;
+import fr.univtln.lhd.model.entities.slots.Subject;
+import fr.univtln.lhd.model.entities.users.Professor;
+import lombok.extern.slf4j.Slf4j;
+import org.threeten.extra.Interval;
+
+import java.sql.*;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+@Slf4j
+/*
+ */
+public class SlotDAO implements DAO<Slot> {
+    private SlotDAO () {
+    }
+
+    private static final String SELECT_STMT = """
+            select id,date_part('epoch',lower(timerange)) as begin_epoch,
+            date_part('epoch',upper(timerange)) as end_epoch,classroom,memo,subject,type
+            from slots WHERE id = (?)
+            """;
+    //private final String save = "INSERT INTO slots (DEFAULT,(?),(?),(?),(?),(?)";
+    //private final String  update;
+
+    private  static final String DELETE_STMT = "DELETE FROM slots WHERE id=(?) CASCADE";
+
+    public static SlotDAO getInstance () {
+        return new SlotDAO();
+    }
+
+
+    /**
+     * Getter for one Slot
+     *
+     * @param id numerical long identifier for getting the Slot
+     * @return May return one Slot instance
+     */
+    @Override
+    public Optional<Slot> get ( final long id ) throws SQLException {
+        try (Connection conn = Datasource.getConnection();
+             PreparedStatement stmt = conn.prepareStatement(SELECT_STMT)
```

Comment:
Why doing the prepared statement now, and not when the dao is create ?

Reply:
Hmm, correct me if I am wrong,but my comprehension is that if you do a prepared satement
you got one more connnection to the db
```
DAO                 DB
|send prepared stm   |
|------------------->|
|return compile stm  |
|<-------------------|
|                    |
.....after............
|send compile stm    |
|------------------->|
|return result set   |
|<-------------------|

```
I assume that we don't have reason to fear an sql injection, therefore would not it be better to do a simple statement ?

Reply:
The setString, setInt and other methods are here to prevent SQL injections. 
### TLDR: here is a statement analogy
> Hey postgres I'm gonna need some id, timestamp, and thing from a table, I'll give you the values  later.
 Postgres can anticipate, as the query planning rarely needs the specific IDs or values.  
### Details 
As the java documentation says : 

The process is as follows : 
1. A connection is opened, in our case, it's just a pop from the set (ConcurrentBag) of HikariCP
2. The String is sent to the database, using a `PREPARE `command, which as per postgresQL documentation [^1]  just sends the String with a placeholder (like the `?` of JDBC)  for the settings, this allows the DBMS to plan the query(ies) as it has the information on rows and columns at hand. This is useful, if part of the parameters in your query rely on values that need time to be computed.
 


4. Once the statement's results are processed, the connection is closed (given back to the pool), when the try block is complete

[^1]:Text above from [PostgreSQL docs](https://www.postgresql.org/docs/15/sql-prepare.html) and [Java docs](https://docs.oracle.com/javase/tutorial/jdbc/basics/prepared.html), respectively
> When the PREPARE statement [statement here means SQL language instruction] is executed, the specified statement is parsed, analyzed, and rewritten. When an EXECUTE [in java ExecuteQuery()] command is subsequently issued, the prepared statement is planned and executed. This division of labor avoids repetitive parse analysis work, while allowing the execution plan to depend on the specific parameter values supplied.


> The main feature of a PreparedStatement object is that, unlike a Statement object, it is given a SQL statement when it is created. The advantage to this is that in most cases, this SQL statement is sent to the DBMS right away, where it is compiled. As a result, the PreparedStatement object contains not just a SQL statement, but a SQL statement that has been precompiled. This means that when the PreparedStatement is executed, the DBMS can just run the PreparedStatement SQL statement without having to compile it first.